lib/openshift/network.rb

```diff
@@ -0,0 +1,35 @@
+require 'openshift/cluster_resource'
+
+module BushSlicer
+  # @note represents an OpenShift environment Network class
+  class Network < ClusterResource
+
+    def cluster_network(user: nil, cached: true, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      rr.dig('spec', 'clusterNetwork')
+    end
+
+  end
+
+  class NetworkOperator < Network
+    RESOURCE = "networks.operator.openshift.io"
+    def network_type(user: nil, cached: true, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      rr.dig('spec', 'defaultNetwork')["type"]
```

Comment:
if using `dig` it makes more sense to do `rr.dig('spec', 'defaultNetwork', "type")` Otherwise you can get into no method for nil class error.